forge/src/executor/fork/mod.rs

```diff
@@ -0,0 +1,461 @@
+        let provider = self.provider.clone();
+        let block_id = self.block_id;
+        let fut = Box::pin(async move {
+            let balance = provider.get_balance(address, block_id);
+            let nonce = provider.get_transaction_count(address, block_id);
+            let code = provider.get_code(address, block_id);
+            let resp = tokio::try_join!(balance, nonce, code);
+            (resp, address)
+        });
+        ProviderRequest::Account(fut)
+    }
+
+    /// process a request for an account
+    fn request_account(&mut self, address: Address, listener: AccountListener) {
+        match self.account_requests.entry(address) {
+            Entry::Occupied(mut entry) => {
+                entry.get_mut().0.push(listener);
+            }
+            Entry::Vacant(entry) => {
+                entry.insert((vec![listener], Default::default()));
+                self.pending_requests.push(self.get_account_req(address));
+            }
+        }
+    }
+}
+
+impl<M> Future for BackendHandler<M>
+where
+    M: Middleware + Clone + Unpin + 'static,
+{
+    type Output = ();
+
+    fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        let pin = self.get_mut();
+
+        // receive new requests to delegate to the underlying provider
+        while let Poll::Ready(Some(req)) = Pin::new(&mut pin.incoming).poll_next(cx) {
+            pin.on_request(req)
+        }
+
+        // poll all requests in progress
+        for n in (0..pin.pending_requests.len()).rev() {
+            let mut request = pin.pending_requests.swap_remove(n);
+            match &mut request {
+                ProviderRequest::Account(fut) => {
+                    if let Poll::Ready((resp, addr)) = fut.poll_unpin(cx) {
+                        // get the response
+                        let (balance, nonce, code) = resp.unwrap_or_else(|_| {
+                            tracing::trace!("Failed to get account for {}", addr);
+                            Default::default()
+                        });
+
+                        // conver it to revm-style types
+                        let (code, code_hash) = if !code.0.is_empty() {
+                            (Some(code.0.clone()), keccak256(&code).into())
+                        } else {
+                            (None, KECCAK_EMPTY)
+                        };
+
+                        // TODO: Is the storage stuff here still required?
+                        let (listeners, storage) =
+                            pin.account_requests.remove(&addr).unwrap_or_default();
+                        let acc = AccountInfo { nonce: nonce.as_u64(), balance, code, code_hash };
+                        pin.cache.write().insert(addr, (acc.clone(), storage));
+                        // notify all listeners
+                        for listener in listeners {
+                            match listener {
+                                AccountListener::Basic(sender) => {
```

Comment:
~~iirc `Basic` was nomenclature of sputnik, perhaps we rename that as well? `account(_info)`?~~
nvm there's still `Database::basic`